lblod/ember-rdfa-editor
# lists: remove top margin from div elements inside li elements

### Overview
The changes in the rdfa-representations of our documents caused an (unwanted) side-effect. As `li`elements may now have `div` elements inside of them (to store the RDF data), a unwanted visual change was introduced.
Elements which appear after `div` elements inside an editor document have a `margin-top` of 1.2rem applied to them. (https://github.com/lblod/ember-rdfa-editor/blob/43ef4ba006d18626d674a0d308579b706069d628/app/styles/ember-rdfa-editor/_c-content.scss#L122-L130)

In order to represent their RDF data, `li` tags now contain two consecutive `div` elements:
- 1 `div` element to store the 'hidden' RDF
- 1 `div` element to store the content of the list item

The second `div` element now gets a `margin-top` applied, causing an unexpected visual regression.

This PR contains a possible solution: not applying the `margin-top` rule when the `div` elements are inside an `li` element. A similar solution was already used for `p` elements inside list items. This may however not be the best solution.

Additionally this PR also slightly modifies a css rule used for inheriting list-style-types.

##### connected issues and PRs:
None

### How to test/reproduce
- `pnpm start`
- Create a list with multiple levels
- Notice that no additional margin is applied to the `div` elements inside the `li` elements


### Challenges/uncertainties
The solution include in this PR may not be the best solution. As the first `div` element (of the two `div` elements present inside a list item) is not visible, it is not very logical to apply a `top-margin` in this case. We should probably add a `say-hidden` class/data-attribute to hidden element in order to ensure they do not count towards margins.

### Checks PR readiness
